### PR TITLE
feat: mailer logging

### PR DESCRIPTION
**Pull request by cstockton**
This is a quick patch to add basic logging information to the mailer package.

I wasn't sure how to only log this information when using the Supabase default mail provider without adding another custom config flag to enable it.

### CODE REVIEW

File: internal/api/middleware.go [hunk 1]
```diff
@@ -173,7 +173,7 @@ func (a *API) isValidAuthorizedEmail(w http.ResponseWriter, req *http.Request) (
 			}
 		}
 
-		return ctx, badRequestError(ErrorCodeEmailAddressNotAuthorized, "Email address %q cannot be used as it is not authorized", email)
+		return ctx, badRequestError(ErrorCodeEmailAddressNotAuthorized, "Email address %q is not a member of this organizations team.", email)
```
Comment (kangmingtay) [inline internal/api/middleware.go:176]: the error message shouldn't contain information about the wider Supabase jargon like "members" or "organizations" - we have some folks who self-host the auth service outside of the Supabase ecosystem

File: internal/mailer/mailme.go [hunk 2]
```diff
@@ -82,8 +84,45 @@ func (m *MailmeMailer) Mail(to, subjectTemplate, templateURL, defaultTemplate st
...
+		if len(val) == 0 || val[0] == "" {
+			continue
+		}
+
+		hdr := strings.ToLower(key)
+		if hdr == mailSupabaseProjectRefHeader {
+			return val[0], true
+		}
+	}
+	return "", false
```
Comment (kangmingtay) [inline internal/mailer/mailme.go:125]: likewise, i don't think we should leak the concept of a "project ref" to the auth service

File: internal/api/middleware.go (hunk 1 above)
Comment (hf) [reply internal/api/middleware.go:176]: I'll fix this in the infra.

Comment (hf) [reply internal/api/middleware.go:176]: https://github.com/supabase/infrastructure/pull/20244

File: internal/mailer/mailme.go [hunk 3]
```diff
@@ -82,8 +84,45 @@ func (m *MailmeMailer) Mail(to, subjectTemplate, templateURL, defaultTemplate st
...
+			return
+		}
+
+		fields := logrus.Fields{
+			"event":         "mail.send",
+			"project_ref":   projectRef,
+			"mail_from":     m.From,
+			"mail_to":       to,
+			"mail_subject":  subject,
+			"mail_template": subjectTemplate,
```
[the remark this answers is not in the document]
Comment (cstockton) [reply internal/mailer/mailme.go:100]: I can remove the `mail_subject`, I think the "subjectTemplate" is the "invite" "confirm" value isn't it?

**Pull request merged**